### PR TITLE
updated Timber dependency version to 4.1.2

### PR DESCRIPTION
**Pull request by mmirhoseini**
this old Timber version causes me many troubles,
so please merge this new version fix

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

File: stetho-timber/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -7,9 +7,13 @@
...
 
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-          package="com.facebook.stetho.timber">
+<manifest
+    xmlns:tools="http://schemas.android.com/tools"
+    package="com.facebook.stetho.timber">
 
-    <application />
+    //fixes Timber Library min sdk issue
+    <uses-sdk tools:overrideLibrary="timber.log"/>
```
Comment (jasta) [inline stetho-timber/src/main/AndroidManifest.xml]: It seems like this entry is the only thing that would actually fix your issue.  Why are there so many unrelated changes (like bumping our build tools version)?

Comment (mmirhoseini) [inline stetho-timber/src/main/AndroidManifest.xml]: Hi, you are correct, but these changes cause failure in travis build, and I try to fix it but I couldn't.

**Comment by jasta** [comment]
I'll separate out the unrelated chnages myself as I need them to update my Android Studio configuration.  Then we can merge this in after.  I'm also going to try one more time to fix the Travis CI issues (they're caused by stetho-js-rhino being too large and failing to dx on their servers, *sigh*).

**Comment by jasta** [comment]
Can you rebase to enable auto merging then we should be good.

**Comment by mmirhoseini** [comment]
I will rebase and submit another pull request.
